whylog/config/__init__.py

```diff
@@ -149,21 +181,27 @@ def get_parsers_for_log_type(self, log_type):
 
 
 class InvestigationStep(object):
-    """
-    Represents rules, parsers and locations of logs which are necessary
-    to find and parse log files with potential causes.
-    """
-
-    def __init__(self, parsers, rules, log_location, effect_time):
-        pass
-
-    def get_clues(self, line):
+    def __init__(self, concatenated_parser, effect_time, earliest_cause_time):
+        self._concatenated_parser = concatenated_parser
+        self.effect_time = effect_time
+        self.earliest_cause_time = earliest_cause_time
+
+    # mocked Clue for second line in node_1.log for 003 test
+    def mocked_clues(self):
+        line_source = LineSource('localhost', 'node_1.log', 40)
+        line_time = datetime(2015, 12, 3, 12, 8, 8)
+        regex_parametes = {'cause': (line_time,)}
+        return {
+            'cause': Clue(regex_parametes, line_time, '2015-12-03 12:08:08 root cause', line_source)
+        }
```

Comment:
In the docstring of function `get_clues` you wrote: ":returns: list of created clues". But here your code returns a dict instead of list. Please change it to list or discuss with me about design.